cyctl/internal/create/modules.go

```diff
@@ -74,7 +86,12 @@ var (
 		Args:    cobra.ExactArgs(1),
 		Aliases: []string{"modules"},
 		Run: func(cmd *cobra.Command, args []string) {
-			createModule(kubeconfig.Moduleset, args[0], repo, path, version, namespace, valuesFile)
+			// Custom validation:
+			// Either templateName or (repo and path) must be provided, if one is provided the other must be empty
+			if (templateName != "" && (repo != "" || path != "")) || (templateName == "" && (repo == "" || path == "")) {
+				log.Fatalf("Error: Either template or (repo and path) must be provided.")
+			}
```

Comment:
Can you also add version here?
```suggestion
			if (templateName != "" && (repo != "" || path != "" || version != "")) || (templateName == "" && (repo == "" || path == "")) {
				log.Fatalf("Error: Either template or (repo, path and version) must be provided.")
			}
```

Reply:
Version was not a compulsory flag that is why I left it. I have added it now.

Reply:
Makes sense @shivamrazorpay. This is just users dont use the --version flag to override the version in the template.
Can you push the update?